.github/workflows/tests.yaml

```diff
@@ -462,20 +462,8 @@ jobs:
                       --target linux-x64-java-matter-controller \
                       build \
                    "
-            - name: Run Discover Tests
-              timeout-minutes: 65
-              run: |
-                  scripts/run_in_build_env.sh \
-                  './scripts/tests/run_java_test.py \
-                     --app out/linux-x64-all-clusters-ipv6only-no-ble-no-wifi-tsan-clang-test/chip-all-clusters-app \
-                     --app-args "--discriminator 3840 --interface-id -1" \
-                     --tool-path out/linux-x64-java-matter-controller \
-                     --tool-cluster "discover" \
-                     --tool-args "commissionables" \
-                     --factoryreset \
-                  '
```

Comment:
Why disable the discovery test? This test does not check if the commission-able devices can be found from the environment, but it check if the Java API discoverCommissionableNodes has been called successfully, an exception will be thrown in case of failure, which should in turn to cause the java-matter-controller to call "System.exit(1)"

Reply:
This is failing

Reply:
[2023-01-12 18:43:41.952013][JAVA ][STDOUT][45937:45940] CHIP:IN: TransportMgr initialized
[2023-01-12 18:43:41.952142][JAVA ][STDOUT][45937:45940] CHIP:CTL: Setting up group data for Fabric Index 1 with Compressed Fabric ID:
[2023-01-12 18:43:41.952212][JAVA ][STDOUT][45937:45940] CHIP:SPT: 0x7c, 0x00, 0x09, 0x1c, 0xc1, 0x05, 0xd0, 0xdb, 
[2023-01-12 18:43:41.959577][JAVA ][STDOUT][45937:45954] CHIP:CTL: IO thread starting
[2023-01-12 18:43:49.001990][JAVA ][STDOUT]Run command failed with exception: Conversion = 'l'
[2023-01-12 18:43:49.043335][JAVA ][STDERR]Jan 12, 2023 6:43:49 PM com.matter.controller.commands.common.CommandManager showCommand
[2023-01-12 18:43:49.043405][JAVA ][STDERR]INFO: Usage:
[2023-01-12 18:43:49.044414][JAVA ][STDERR]Jan 12, 2023 6:43:49 PM com.matter.controller.commands.common.CommandManager showCommand
[2023-01-12 18:43:49.044442][JAVA ][STDERR]INFO: 
[2023-01-12 18:43:49.044450][JAVA ][STDERR][--paa-trust-store-path]:
[2023-01-12 18:43:49.044474][JAVA ][STDERR]  Path to directory holding PAA certificate information.  Can be absolute or relative to the current working directory.
[2023-01-12 18:43:49.044481][JAVA ][STDERR]
[2023-01-12 18:43:49.044485][JAVA ][STDERR][--cd-trust-store-path]:
[2023-01-12 18:43:49.044492][JAVA ][STDERR]  Path to directory holding CD certificate information.  Can be absolute or relative to the current working directory.
[2023-01-12 18:43:49.044496][JAVA ][STDERR]
[2023-01-12 18:43:49.044515][JAVA ][STDERR][--commissioner-name]:
[2023-01-12 18:43:49.044541][JAVA ][STDERR]  Name of fabric to use. Valid values are "alpha", "beta", "gamma", and integers greater than or equal to 4.  The default if not specified is "alpha".
[2023-01-12 18:43:49.044548][JAVA ][STDERR]
[2023-01-12 18:43:49.044551][JAVA ][STDERR][--commissioner-nodeid]:
[2023-01-12 18:43:49.044557][JAVA ][STDERR]  The node id to use for java-matter-controller.  If not provided, kTestControllerNodeId (112233, 0x1B669) will be used.
[2023-01-12 18:43:49.044561][JAVA ][STDERR]
[2023-01-12 18:43:49.044564][JAVA ][STDERR][--use-max-sized-certs]:
[2023-01-12 18:43:49.044570][JAVA ][STDERR]  Maximize the size of operational certificates. If not provided or 0 ("false"), normally sized operational certificates are generated.
[2023-01-12 18:43:49.044575][JAVA ][STDERR]
[2023-01-12 18:43:49.044579][JAVA ][STDERR][--only-allow-trusted-cd-keys]:
[2023-01-12 18:43:49.044585][JAVA ][STDERR]  Only allow trusted CD verifying keys (disallow test keys). If not provided or 0 ("false"), untrusted CD verifying keys are allowed. If 1 ("true"), test keys are disallowed.
[2023-01-12 18:43:49.044590][JAVA ][STDERR]
[2023-01-12 18:43:49.046539][JAVA ][STDOUT]  java-matter-controller discover commissionables [--paa-trust-store-path] [--cd-trust-store-path] [--commissioner-name] [--commissioner-nodeid] [--use-max-sized-certs] [--only-allow-trusted-cd-keys]

Reply:
Could we debug/fix this as well as part of this PR? Or is the error cause unclear at this time?

Instead of deletion, how about setting `if: false` (if that is supported) with a comment of adding back once we fix the bug. Please create a bug in github for this.

https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution

Reply:
Oops, it looks we have regression and caught  by the CI, the Java API should not fail, I will take a look

Reply:
https://github.com/project-chip/connectedhomeip/pull/24007, I think this first PR does not pass this test. 
I would setting if else here. Yufeng would continue to debug it further.

Reply:
https://github.com/project-chip/connectedhomeip/issues/24411